src/main/scanLogic/abstractScanLogic.ts

```diff
@@ -15,7 +14,7 @@ export abstract class AbstractScanLogic {
      */
     public abstract scanAndCache(
         progress: vscode.Progress<{ message?: string; increment?: number }>,
-        componentsToScan: Set<ComponentDetails>,
+        componentsToScan: Components[],
```

Comment:
If I'm understanding right -
This triples the amount of memory consumed in the scanning (which is already very high) - 
instead of utilizing 1 string of component ID, we store also the project path and project type. 
The project path and types are needed only for the root nodes.